tests/functional/api/test_merge_requests.py

```diff
@@ -181,11 +181,27 @@ def test_merge_request_reset_approvals(gitlab_url, project, wait_for_sidekiq):
     assert mr.reset_approvals()
 
 
+def test_cancel_merge_when_pipeline_succeeds(project, merge_request, wait_for_sidekiq):
+    mr = merge_request(source_branch="test_merge_request_merge", create_pipeline=True)
+    # Set to merge when the pipeline succeeds, which should never happen
+    mr.merge(merge_when_pipeline_succeeds=True)
+    wait_for_sidekiq(timeout=60)
+
+    mr = project.mergerequests.get(mr.iid)
```

Comment:
A probably newbie question, do we need to get a handle of the mr even though it was already done in line 185?

Reply:
This is just to fetch the latest actual state from the server with all the attributes after performing the merge I'd say.